### PR TITLE
Rpc blobs bug fix + blobsUtil endianness + integration tests

### LINKED ISSUE #6834: Add BlobSidecarsByRangeIntegrationTest
Add `BlobSidecarsByRangeIntegrationTest` when the syncing and the storage are done

### LINKED ISSUE #7183: BlobSidecarsByRange does not work
Running interop, testing forward sync I realize blobs by range RPC doesn't work.

- the request is sent from the client.
- it is received from the server and the server prepares the response.
- As soon as the server start streaming the result to client, client closes the connection falling in this condition:
in `Eth2OutgoingRequestHandler#processData`
image

- Server early stops the streaming this way:
image

### LINKED ISSUE #7216: `BlobSidecarByRoot` RPC should not fail if block root is not found.
```
Requests sidecars by block root and index.
The response is a list of BlobSidecar whose length is less than or equal to the number of requests.
It may be less in the case that the responding peer is missing blocks or sidecars.
```

### PR DESCRIPTION
**Pull request by tbenr**
fixes: #7216
fixes: #7183
fixes: #6834

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRootMessageHandler.java [hunk 1]
```diff
@@ -164,25 +156,29 @@ private UInt64 getFinalizedEpoch() {
    *   <li>The block root references a block greater than or equal to the minimum_request_epoch
```
Comment (StefanBratanov) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRootMessageHandler.java:155]: I would just remove the `<li>A block for the block root is available` above

**Review by StefanBratanov — APPROVED** [review]
LGTM with one javadoc nit

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DefaultEth2Peer.java [hunk 2]
```diff
@@ -290,9 +293,19 @@ public SafeFuture<Void> requestBlobSidecarsByRange(
                 if (updatedCount.isZero()) {
                   return SafeFuture.COMPLETE;
                 }
-                request = new BlobSidecarsByRangeRequestMessage(firstSupportedSlot, updatedCount);
+
+                request =
+                    new BlobSidecarsByRangeRequestMessage(
+                        firstSupportedSlot,
+                        updatedCount,
+                        spec.getMaxBlobsPerBlock(firstSupportedSlot.plus(updatedCount))
```
Comment (StefanBratanov) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DefaultEth2Peer.java]: should be technically` .plus(updatedCount).minus(1)`

**Review by StefanBratanov — COMMENTED** [review]
LGTM with one javadoc nit

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DefaultEth2Peer.java (hunk 2 above)
Comment (tbenr) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DefaultEth2Peer.java]: right.. good point

Comment (tbenr) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DefaultEth2Peer.java]: put it on the other place too

**Pull request merged**